pkg/parser/parser.go

```diff
@@ -576,6 +590,10 @@ func (p *Parser) advanceTo(pos int) {
 	p.pos = pos
 	p.cur = p.lookAt(pos)
 	p.peek = p.lookAt(pos + 1)
+	inc := 2
+	for p.peek.Type == lexer.WS {
```

Comment:
yuck.

Reply:
Is this yuck because you don't advance `inc` so the loop never terminates if there are two consecutive whitespace tokens? I know there can't be because they are folded, but then why the loop?

Reply:
yeah you are right. WS is collapsed into a single token so I should just advance one more if it is WS. I will update this.

Reply:
ok, I replaced the loops with if's. That feels a little less yuck I guess.

I don't think I can just assign `cur = peek` as I also need to know the position (of peek), I could track the peekPos, but not sure if that makes things easier...